### ISSUE TITLE
testautomation fails to link with a Ubuntu MinGW toolchain

### ISSUE THREAD

**Issue by madebr** [libsdl-org/SDL#9296]
https://github.com/madebr/SDL/actions/runs/8319388990/job/22762942004#step:7:461
```
 [ 39%] Linking C executable testautomation.exe
cd /home/runner/work/SDL/SDL/build_x86/SDL_test && /usr/local/bin/cmake -E cmake_link_script CMakeFiles/testautomation.dir/link.txt --verbose=1
/usr/local/bin/cmake -E rm -f CMakeFiles/testautomation.dir/objects.a
/usr/bin/i686-w64-mingw32-ar qc CMakeFiles/testautomation.dir/objects.a @CMakeFiles/testautomation.dir/objects1.rsp
/usr/bin/i686-w64-mingw32-gcc -O3 -DNDEBUG -Wl,--whole-archive CMakeFiles/testautomation.dir/objects.a -Wl,--no-whole-archive -o testautomation.exe -Wl,--out-implib,libtestautomation.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles/testautomation.dir/linkLibs.rsp
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/testautomation.dir/objects.a(testautomation_intrinsics.c.obj):testautomation_intrinsics.c:(.text+0x71d): undefined reference to `__builtin_ia32_crc32si'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/testautomation.dir/objects.a(testautomation_intrinsics.c.obj):testautomation_intrinsics.c:(.text+0x73e): undefined reference to `__builtin_ia32_crc32hi'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/testautomation.dir/objects.a(testautomation_intrinsics.c.obj):testautomation_intrinsics.c:(.text+0x75b): undefined reference to `__builtin_ia32_crc32qi'
collect2: error: ld returned 1 exit status
```
These intrinsics are called [here](https://github.com/libsdl-org/SDL/blob/12245e4c756eb964ecdf4a528c36aab6f971baf5/test/testautomation_intrinsics.c#L252-L281) as an example of sse4.2 intrinsics.

**Comment by sezero** [comment]
Broken toolchain?

**Comment by madebr** [comment]
I think so, I'm building the tests with sse42 disabled.

**Comment by madebr** [comment]
How come the `SDL_SSE_4_2` test in `CMakeLists.txt` does not catch it?
Perhaps we need to add some tests to it?
Or alternatively, is the automation test too complicated?

**Comment by sezero** [comment]
I don't know the cmake'ry involved.

The thing is building with gcc-7.3.1 for linux doesn't fail, which is baffling.

**Comment by madebr** [comment]
I'm suggesting to add tests here until it fails:
https://github.com/libsdl-org/SDL/blob/384e5486bc6ee5d0ae9f9c2c7a25d660b4823582/CMakeLists.txt#L752-L764

Configure with `-UCOMPILER_SUPPORTS_SSE4_2` to force a retest while iterating.

**Comment by sezero** [comment]
OK, found it: Looking at the preprocessed source, intrin.h of mingw-w64
headers used to _prototype_ those three builtins which is the source of
the issue. Removing them fixes the issue. Mainstream fixed it in v9.0.0:
https://github.com/mingw-w64/mingw-w64/commit/62924d0cd8ea3422689023a5be546755fe73cbe0
